common/dynamicconfig/config_test.go

```diff
@@ -323,6 +324,18 @@ func TestDynamicConfigFilterTypeIsMapped(t *testing.T) {
 	}
 }
 
+func TestDynamicConfigFilterTypeIsParseable(t *testing.T) {
+	for idx, filterString := range filters { // TestDynamicConfigFilterTypeIsMapped ensures this is a complete list
+		if idx == 0 {
+			assert.Equalf(t, UnknownFilter, ParseFilter(filterString), "first filter string should have parsed as unknown: %v", filterString)
+			// unknown filter string is likely safe to change and then should be updated here, but otherwise this ensures the logic isn't entirely position-dependent.
+			require.Equalf(t, "unknownFilter", filterString, "expected first filter to be 'unknownFilter', but it was %v", filterString)
+		} else {
+			assert.NotEqualf(t, UnknownFilter, ParseFilter(filterString), "failed to parse filter: %s, make sure it is in ParseFilter's switch statement", filterString)
```

Comment:
this is a weak validation that only checks it's not parsed as UnknownFilter but better than nothing

Reply:
a test that checks that the mapping is what we expect would just be a re-implementation of the func itself :\  though I can check that it's unique, since nothing enforces that currently.

I think this would all probably be better done with a map rather than a slice, so the pairing can be built up in a single hardcoded location, I just kinda don't want to make major changes here in this PR.

Reply:
added a uniqueness check too

Reply:
that's fair. without changing `filters` into a map I don't see a future proof way. it can be handled separately since this PR is quite big already